### ISSUE TITLE
Robbing The Fed produces 0 hits on custom model

### ISSUE THREAD

**Issue by anirban-nath** [JonasGeiping/breaching#9]
My model is a multi-output model where I combine losses from segmentation and classification for final loss and back-prop. I have been working to get my model fitted to Robbing the Fed attack. All my parameters seem the same as that of the given example except that I am performing segmentation+classification instead of just classification and my images are 1x512x512 (single channel images) instead of 3x224x224. 

What could be the reason that I am not getting any hits?

edit: I was comparing my model with the one that is used in the example file and I noticed that my model.buffer values are very very large, some going upto 10^9. What could be the reason for this?

**Comment by JonasGeiping** [comment]
Hi, 

can you provide more details how you constructed your model? Where is the ImprintBlock placed? Also, what buffers are we talking about BatchNorm values? What about the gradients, are those large too?

**Comment by anirban-nath** [comment]
Hi, my model is a multi-task model that calculates losses for segmentation and classification, sums them up, and backpropagates through the whole model. I use a loss function that is a sum of Dice and Cross Entropy.

> Where is the ImprintBlock placed? 

The ImprintBlock is placed before the first layer (default setting) which is a Conv2D layer.

> Also, what buffers are we talking about; BatchNorm values?

Yes. For every BatchNorm layer, I have a set of 3 buffers, for example: `center.bn1.running_mean', 'center.bn1.running_var', 'center.bn1.num_batches_tracked`

> What about the gradients, are those large too?

No, the gradient and weights values are pretty normal, mostly between -1 and 1.

**Comment by JonasGeiping** [comment]
Hm, that is surprising -  there should not be any interaction with batchnorm parameters in the following layers. Does this happen with both losses, or does it go away if one of them is turned off?

**Comment by JonasGeiping** [comment]
Closing this for now. Let me know if you have more questions.
